### ISSUE TITLE
WPF Scottplot5 How to change the scale interval?

### ISSUE THREAD

**Issue by xinRedi** [ScottPlot/ScottPlot#4027]
I have disabled the zoom function, but how do I set the scale interval value? Like interval value 50? Or 300?
![微信图片_20240702105721](https://github.com/ScottPlot/ScottPlot/assets/82932045/5e1f41d9-c17b-4cf3-bbd7-f241135b246c)

**Comment by KroMignon** [comment]
hello @xinRedi, did you try to use `SetAxisLimit()`? => https://scottplot.net/cookbook/5.0/AxisAndTicks/SetAxisLimits/

If you only want to set vertical axis, you could also user `SetLimitsY()`:
```
ScottPlot.Plot myPlot = new();

myPlot.Add.Signal(Generate.Sin(51));
myPlot.Add.Signal(Generate.Cos(51));

// auto scale the horizontal axis
myPlot.Axes.AutoScaleX(); 
// force the vertical axis limits
myPlot.Axes.SetLimitsY(0, 5);

myPlot.SavePng("demo.png", 400, 300);
```

**Comment by xinRedi** [comment]
> 你好，你试过用吗？=> https://scottplot.net/cookbook/5.0/AxisAndTicks/SetAxisLimits/`SetAxisLimit()`
> 
> 如果只想设置垂直轴，也可以使用：`SetLimitsY()`
> 
> ```cs
> ScottPlot.Plot myPlot = new();
> 
> myPlot.Add.Signal(Generate.Sin(51));
> myPlot.Add.Signal(Generate.Cos(51));
> 
> // auto scale the horizontal axis
> myPlot.Axes.AutoScaleX(); 
> // force the vertical axis limits
> myPlot.Axes.SetLimitsY(0, 5);
> 
> myPlot.SavePng("demo.png", 400, 300);
> ```

你好，我不是设置起始值与终止值，我要设置间隔值

**Comment by KroMignon** [comment]
@xinRedi sorry I don't understand Chinese, I hope DeepL translation is correct:
"Hi, I'm not setting the start and end values, I'm setting the interval values."

If you want to define the tick steps, you could use `ScottPlot.TickGenerators` => https://scottplot.net/cookbook/5.0/CustomizingTicks/AltTickGen/
```csharp
ScottPlot.Plot myPlot = new();

myPlot.Add.Signal(Generate.Sin(51, 24));
myPlot.Add.Signal(Generate.Cos(51, 36));

// auto scale the horizontal axis
myPlot.Axes.AutoScale(); 
// force the vertical steps
myPlot.Axes.Left.TickGenerator = new ScottPlot.TickGenerators.NumericFixedInterval(7);

myPlot.SavePng("demo.png", 400, 300);
```
![demo](https://github.com/ScottPlot/ScottPlot/assets/3978902/666e5777-7098-466a-a2e2-afe76430c314)

**Comment by xinRedi** [comment]
Sorry, I forgot to translate, thank you, my problem is solved, but there is a new problem, my start value is -200, the end value is 1800, NumericFixedInterval(100), why is the start value not displayed?
![捕获](https://github.com/ScottPlot/ScottPlot/assets/82932045/be5681fa-d502-4243-b707-0b670e475ccd)

**Comment by xinRedi** [comment]
> 对不起，我不懂中文，我希望DeepL的翻译是正确的：“嗨，我不是在设置开始值和结束值，而是在设置区间值。
> 
> 如果要定义刻度步骤，可以使用 => https://scottplot.net/cookbook/5.0/CustomizingTicks/AltTickGen/`ScottPlot.TickGenerators`
> 
> ```cs
> ScottPlot.Plot myPlot = new();
> 
> myPlot.Add.Signal(Generate.Sin(51, 24));
> myPlot.Add.Signal(Generate.Cos(51, 36));
> 
> // auto scale the horizontal axis
> myPlot.Axes.AutoScale(); 
> // force the vertical steps
> myPlot.Axes.Left.TickGenerator = new ScottPlot.TickGenerators.NumericFixedInterval(7);
> 
> myPlot.SavePng("demo.png", 400, 300);
> ```
> 
> ![演示](https://private-user-images.githubusercontent.com/3978902/345705683-666e5777-7098-466a-a2e2-afe76430c314.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNDM2MDgsIm5iZiI6MTcyMDE0MzMwOCwicGF0aCI6Ii8zOTc4OTAyLzM0NTcwNTY4My02NjZlNTc3Ny03MDk4LTQ2NmEtYTJlMi1hZmU3NjQzMGMzMTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDVUMDEzNTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjFjYzg2MjdjN2VjNGE5M2Y5ZGYwOTcxMTE1MTI3OGQyMWZmOTQ1YzI2MmMxMjk4MDgzYjc3MDIwNmM3YmIyMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.qIGMnqib6T7F1TxHAksIS45tsKj3-RZ-d146ccnlht0)

This is the way I can think of, is there any other convenient way?
![捕获1](https://github.com/ScottPlot/ScottPlot/assets/82932045/8dd2b3aa-6ad3-426a-a673-767e042ce448)
![捕获2](https://github.com/ScottPlot/ScottPlot/assets/82932045/a7066bdb-e67e-4302-9986-4f909693ab1d)

**Comment by epegeot** [comment]
Sorry to be upping this issue, this has been fixed in ScottPlot 5.0.37, you can close it if you don't have further questions on this topic
